kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -111,11 +110,17 @@ class BatchJobSubmission(session: KyuubiBatchSessionImpl, batchRequest: BatchReq
   }
 
   private def submitBatchJob(): Unit = {
+    var applicationStatus: Option[Map[String, String]] = None
+    var appStatusFirstUpdated = false
     try {
       info(s"Submitting $batchType batch job: $builder")
       val process = builder.start
-      var applicationStatus = currentApplicationState
+      applicationStatus = currentApplicationState
       while (!applicationFailed(applicationStatus) && process.isAlive) {
+        if (!appStatusFirstUpdated && applicationStatus.isDefined) {
+          session.sessionManager.sessionStateStore.updateBatchAppInfo(batchId, applicationStatus)
+          appStatusFirstUpdated = true
```

Comment:
shall we put update before the loop if we only want to update it once.

Reply:
we have to use a loop to wait the applicationStatus is non-empty